### ISSUE TITLE
Production Docker deployment issue

### ISSUE THREAD

**Issue by hatmcgat** [inventree/InvenTree#3860]
Production deployment error

Any help is appreciated. I have read through the docs multiple times, and have decent familiraity with Docker/Docker-Compose and Portainer.

I'm attempting to follow the instructions here:

[https://inventree.readthedocs.io/en/latest/start/docker\_prod/](https://inventree.readthedocs.io/en/latest/start/docker_prod/)

When I attempt any of the commands referenceed:

docker-compose run inventree-server invoke update

docker-compose run inventree-server invoke superuser

docker-compose up -d

I get this response:

Error response from daemon: failed to mount local volume: mount /root/intentree/data:/var/lib/docker/volumes/inventree-production\_inventree\_data/\_data, flags: 0x1000: no such file or directory

I have tried multiple different configs and have tried putting the working directory in several different locations. Based on research I even removed unused volumes to try and avoid conflicts.

I have noted that the local volume path " mount /root/intentree/data" etc... has a type i.e. intentree instead of inventree... 

What I can't figure out is where that value is coming from? I dont have that path referenced in any of the docs. I have tried using the default configs and only changed user name/log in info and the exposed port as 8000 is already in use.

For the record, I did free up 8000 in an attempt to keep everything stock and it threw the same error.

&#x200B;

Note: I have successfully gotten a dev instance up and running per this instruction set:

[https://inventree.readthedocs.io/en/latest/start/docker\_dev/#stop-inventree-services](https://inventree.readthedocs.io/en/latest/start/docker_dev/#stop-inventree-services)

&#x200B;

Thank you in advance!

**Comment by wolflu05** [comment]
Hello @hatmcgat ,

How have you specified the `INVENTREE_EXT_VOLUME` variable in your `.env` file?

**Comment by hatmcgat** [comment]
I have tried multiple configs. 

I uncommented the Linux entry and set it to all of the following on various attempts:

/root/inventree-data
/root/InvenTree/data.
/var/lib/docker/volumes/inventree-production_inventree_data/_data

among others. I also made sure that these file paths existed. I would go to the directory I wanted the data to be stored and would PWD and copy and paste into the .env

The thing I don't understand is where this value "/root/intentree/data" is coming from. It isn't in any of the files (docker-compose.yaml, nginx.prod.conf, nor .env)

This was my first attempt to spin up the production example so I tried to keep it "vanilla"

Thanks!

**Comment by SchrodingersGat** [comment]
@hatmcgat I can't find any reference to "intentree" in the source code, or the documentation. Not sure where that is coming from at all.

Can you share your docker-compose and docker-env files?

**Comment by hatmcgat** [comment]
Good evening. I will share when I get back to that machine. For immediate reference, I used the template files and only changed the port mapping from 8000 to 8002. 

I have attempted various file paths and ran the docker-compose commands out of the current working directory each time. In my previous comment I mentioned the changes I made to the .end file to map the external volume

**Comment by hatmcgat** [comment]
For the record, I'm running Ubuntu Server 18.04 for the OS. If I recall correctly from the docs, bare metal needs 20.xx

**Comment by hatmcgat** [comment]
These are the files I was using. Per the instructions on the docs page and example compose file I didn't change anything on my first several attempts and only passed changes in the .env.

On the first try, I created an InvenTree directory:

/root/InvenTree: ls
data (directory) .env (file) nginx.prod.conf.txt (file) docker-compose.yml

This was the working directory I tried my initial docker-compose commands and produced this error. After that, I rm -rf all the files and directories and moved them to various locations trying again with the same result. I removed any volumes created to try and avoid any conflicts.

[inventree.docker-compose.yml.txt](https://github.com/inventree/InvenTree/files/9881522/inventree.docker-compose.yml.txt)
[inventree.env.txt](https://github.com/inventree/InvenTree/files/9881524/inventree.env.txt)
[inventree.nginx.prod.conf.txt](https://github.com/inventree/InvenTree/files/9881529/inventree.nginx.prod.conf.txt)

**Comment by matmair** [comment]
@hatmcgat I see several possible issues:
- the .env shows the data should be in `/root/inventree-data` does this directory exist and can the docker-user read and write there?
- the .env file declares 1337 as the webport, the nginx-file as 8002, the docker-compose exposes 8002
- the media and static directories are declared as being in `/var/www` in our nginx-file; our .env point to `/root/inventree-data`

Could you try if getting these things aligned helps the situation?

**Comment by hatmcgat** [comment]
Hi @matmair thank you for the follow up.

> @hatmcgat I see several possible issues:
> 
> * the .env shows the data should be in `/root/inventree-data` does this directory exist and can the docker-user read and write there?
**Yes, the docker-user has read and write permission. The directory value was pulled from the output of pwd via copy past into the editor.**
> * the .env file declares 1337 as the webport, the nginx-file as 8002, the docker-compose exposes 8002
**I had assumed the 1337 port in the default was for the worker and server to talk to each other and 8002 would expose the service to the end user. I will change that over in my next attempt.**
> * the media and static directories are declared as being in `/var/www` in our nginx-file; our .env point to `/root/inventree-data`
> **I also thought these default values would get spun up at creation. Are you suggesting that the media and static directories should point to the directory I specify in the .env? In this case, /root/inventree-data?**
> Could you try if getting these things aligned helps the situation?


I'll change these things around and give it another shot today.

**Comment by hatmcgat** [comment]
I believe I have gotten past this issue. I can still not get it deployed successfully, but I'm getting errors in other ways now that I am working through. I ended up changing the values you specified and confirmed the directories were readable and writable. I was able to run 

`docker-compose run inventree-server invoke update`

and

`docker-compose up -d`

The containers all deployed but when accessing the proxy I kept getting "502 Bad Gateway" and the direct path localhost:8002 was connection refused. I'm troubleshooting those now.

I have attached an excerpt of the db logs if anyone is interested in weighing in.

Thank you to the community for the assist!
[_inventree-db_logs.txt](https://github.com/inventree/InvenTree/files/9904188/_inventree-db_logs.txt)

**Comment by matmair** [comment]
@hatmcgat 502 errors generally point to a problem where the proxied application is not running or reachable by the gateway(Nginx).

From the errors in the log it looks like your database is not initialized. Can you try running the update command in a fresh terminal session? If you have an environment setting set that specifies another DB it might a problem. Please post the log of the update command, that might provide more insights into the problem.

**Comment by hatmcgat** [comment]
@matmair thank you for the help. I agree the problem is probably in my Nginx setup (this is the first time I've tried using it) I have previously used Traefik. I'll need to do some more research. I have a feeling it lies in my nginx.conf file. I suspected it had to do with telling the compose file where the conf file was, but when I changed that I got back to volume mount issues.

I'll continue to tinker and refer back to this thread for help. I'm sure it all is user error on my end.
